README.md

```diff
@@ -1,150 +1,235 @@
+1.  Using MicroProfile Config for static configuration injection [LINK]
+2.  Advancing the use of MicroProfile Configuration [LINK]
+
+## Before We Start
+
+Please ensure all of the prerequisites have been installed before continuing with these instructions. Clone down this repository
+https://github.com/OpenLiberty/tutorial-microprofile.git
+and navigate to this repository.
+Change branch to module 3 *tutorial-microprofile*
+*git checkout module4*
+
+### Enabling Static Configuration Properties
+This section will demonstrate how to set the HTTP and HTTPS port values, that the gateway server runs on, as examples of static configuration properties in a static configuration file. 
+1.  Once in the root directory, navigate into the following directory *microservice-gateway/src/main/java/liberty/config* and open the file called *bootstrap.properties*
+This bootstrap.properties file is one of the three preconfigured file spaces in OpenLiberty that can be used for static configuration properties. The other two locations that these static configuration properties can be set are: /META-INF/microprofile-config.properties and the server.env files. Bootstrap.properties file has been used in this example because it allows us to set custom properties that can then 
+2.  Add the following to this file:
+*default.http.port=9080*
+*default.https.port=9081*
+3.  Next, navigate to the *server.xml* file in *microservice-gateway/src/main/liberty/config*. In this file add the feature ```<feature>mpConfig-1.1</feature>```. 
+In the same file, replace the httpEndpoint method with the following code:
+```<httpEndpoint host="*"id="defaultHttpEndpoint"httpsPort="${default.https.port}"httpPort="${default.http.port}"/>```
+This will enable the server.xml to peer into the bootstrap.properties file to retrieve the HTTP and HTTPS port values.
+
+### Creating a Custom Configuration Properties Source
+
+Default config sources (like bootstrap.properties) are static and fixed upon an application starting up, so cannot be modified while the server is running. However, you can externalize configuration data outside the application package so that the configuration changes are updated dynamically. These properties are aptly named "dynamic configuration properties". This section will demonstrate how to set a dynamic configuration property:
+
+1.  The first step in this process is to add a CustomConfig.json file outside of your application. In this tutorial, as an example of where to place this json file, the file path used is: 
+*microservice-gateway/CustomConfigSource.json*
+Feel free to use the same or a similar file path for this file. However, the exact file path is not important as long as the json file lies outside of the application.
+
+2.  In the CustomConfigSource.json file add the following code: 
+
+```{    “config_ordinal”:700,
```

Comment:
You need the 3 backticks on there own line otherwise the formatting will not be correct